### DISCUSSION TITLE
inputProps.onChange is not called for DayPickerInput after selecting a day

### DISCUSSION THREAD

**Discussion by fc** [gpbl/react-day-picker#1110] [Support]
I would expect `inputProps={{oChange: () => ...}}` to be triggered after selecting a day.

Even looking through the code it definitely looks like it should be executed as can be seen here:
https://github.com/gpbl/react-day-picker/blob/master/src/DayPickerInput.js#L394

I suppose I would expect `onChange` would be triggered before `onDayChange` somewhere around here:
https://github.com/gpbl/react-day-picker/blob/master/src/DayPickerInput.js#L491

For now, I can accomplish it by using `onDayChange` + `parseDate` to get what I want done.

I did not see anyone else reporting this issue to me it seems like a bug or possibly by design.

And, unrelated to this but in the way I am dynamically creating this (I hook up validation in a separate centralized file allowing for dynamic form creation / validation), it would be nice if onDayChange also included the DayPickerInput props so I can use `parseDate` from there instead of having to know which parseDate is being used.

**Comment by gpbl** [comment]
> I would expect `inputProps={{oChange: () => ...}}` to be triggered after selecting a day.

It is meant to work as a normal input `onChange` event, so I've put it before everything happens. What is the reason of your request? 

> it would be nice if `onDayChange` also included the DayPickerInput props so I can use parseDate from there instead of having to know which parseDate is being used.

Yeah I agree we could add a `props` as third argument of `onDayChange` 👍
